### PR TITLE
feat: add otelcol-config binary

### PR DESCRIPTION
**Pull request by amdprophet**
Adds the otelcol-config binary to the installed components.

### CODE REVIEW

File: Dockerfile
Comment (echlebek) [inline Dockerfile:2]: This should probably be opensource-collection-team@sumologic.com

Comment (amdprophet) [reply Dockerfile]: Good catch!

**Comment by amdprophet** [comment]
By default, the packaging code tries to pull required files (e.g. otelcol-sumo & otelcol-config binaries) from the current release on https://github.com/SumoLogic/sumologic-otel-collector/releases. These tests will continue to fail until either a release is cut with the otelcol-config binary or we update this repository to pull the latest artifacts from GitHub CI. I think the latter is a better option for us. We should solve this before this PR is merged.

I manually triggered a package build run to pull the artifacts from GitHub CI and it passes https://github.com/SumoLogic/sumologic-otel-collector-packaging/actions/runs/10428594808.

**Comment by amdprophet** [comment]
In developing SELinux support for the packages I noticed that it seems to be unnecessary. It seems that installing files with a package into standard locations (e.g. `/usr/local/bin/` or `/lib/systemd/system/`) the appropriate labels will automatically be applied. I'd love it if someone else can test this themselves; I used Rocky Linux 9 and had put SELinux into enforcing mode:

```
[amdprophet@localhost ~]$ sestatus
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   enforcing
Mode from config file:          enforcing
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Memory protection checking:     actual (secure)
Max kernel policy version:      33
```

If we do need an `otelcol-sumo-selinux` package, I've done the work on the `feature/selinux` branch. There are a significant amount of changes to support it: https://github.com/SumoLogic/sumologic-otel-collector-packaging/compare/ci/otelcol-config...feature/selinux.

**Comment by amdprophet** [comment]
I opened https://github.com/SumoLogic/sumologic-otel-collector-packaging/pull/107 to unblock the failing tests here.

**Review by ccressent — APPROVED** [review]

**Review by echlebek — APPROVED** [review]

**Pull request merged**
